lib/external-housing-available/external-housing-available.controllers.js

```diff
@@ -1,6 +1,172 @@
+  const geoCoordinatesLib = require('../geo-coordinates/geo-coordinates.controllers')(models).lib;
+  const accountLib = require('../account/account.controllers')(models).lib;
+  const locationLib = require('../location/location.controllers')(models).lib;
+
+
   return {
+    postExternalHousingAvailable: async function(request, h) {
+      /*
+        # Require Account for submitting fields, and location. 
+        # Account also must be SUPER_ADMIN
+      */
+      let account;
+
+      try {
+        account = await accountLib.getAccount(request.auth.credentials);
+      } catch (e) {
+        throw Boom.badImplementation('Error during getAccount(request.auth.credentials).' + e);
+      }
+
+      if (account === null) {
+        throw Boom.badRequest('Must have an Account to post an External Housing Avaliable');
+      }
+
+      if (process.env.SUPER_ADMIN !== request.auth.credentials.subjectId) {
+        throw Boom.forbidden('Must be an Admin Account to post an External Housing Avaliable');
+      }
+
+      //Match address input with geocoder.
+      var geocodes = await postalAddressLib.addressGeocode(request.payload.address);
+      //If address doesnt exist or length is 0. 
+      if (!geocodes.results || geocodes.results.length === 0) {
+        try {
+          //Log it into the logger
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.no_results',
+            severity: 'Notice',
+            request: request,
+          });
+        } catch (e) {}
+        throw Boom.badData('No results for that address');
+      } 
+
+      const firstGeocode = geocodes.results[0];
+
+      let addressObject = postalAddressLib.extractAddress(firstGeocode);
+
+      if (!addressObject.addressNumber) {
+        try {
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.street_number',
+            severity: 'Notice',
+            request: request,
+          });
+        } catch (e) {}
+        throw Boom.badData('Address is missing a street number');
+      }
+
+      //Address without a county, we want to use locality instead.
+      if (!addressObject.hasOwnProperty('addressCounty')) {
+        addressObject.addressCounty = addressObject.addressLocality;
+      }
+      if (
+        addressObject.addressRegion !== process.env.ADDRESS_LIMIT_STATE ||
+        addressObject.addressCounty !== process.env.ADDRESS_LIMIT_COUNTY
+      ) {
+        try {
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.county',
+            severity: 'Notice',
+            request: request,
+          }); 
+        } catch (e) {} 
+        throw Boom.badData('Sorry, We are only accepting properties in ' + `${process.env.ADDRESS_LIMIT_COUNTY}, ${process.env.ADDRESS_LIMIT_STATE}`); 
+      }
+
+      //Extracting the address using function from geoCoordinates lib
+      let geoCoordObject = geoCoordinatesLib.extractLocation(firstGeocode);
```

Comment:
we don't actually use this in this context (do a ctrl+f for "geoCoordObject" and you'll see this is the only result.) so it can be removed.